Source/WebCore/dom/Element.cpp

```diff
@@ -2718,7 +2735,9 @@ void Element::removedFromAncestor(RemovalType removalType, ContainerNode& oldPar
     document().fullscreenManager().exitRemovedFullscreenElementIfNeeded(*this);
 #endif
 
-    if (!hasEffectiveLangState())
+    if (!parentNode() && oldParentOfRemovedTree.isDocumentNode())
+        document().setDocumentElementLanguage(nullAtom());
```

Comment:
Don't we also want to clear hasLangAttrKnownToMatchDocumentElement flag here as well?

Reply:
That flag should never be set on the document element, and an element that is inserted as a new document element will have the flag cleared in Element::clearEffectiveLangStateOnNewDocumentElement.